lib/app.js

```diff
@@ -41,8 +43,29 @@ if (otpConfig.customIcons) {
   })
 }
 
-if (jsConfig.applyCustomIconLogic) {
-  jsConfig.applyCustomIconLogic(customIcons)
+const BaseModeIcon = jsConfig.BaseModeIcon || ClassicModeIcon
+
+// Build a ModeIcon component from a base ModeIcon component
+// and from the customIcons set (they can't be imported from config.js).
+// If there is an exact-matching icon in customIcons, use it.
+const ModeIcon = ({ mode, ...props }) => {
+  if (!mode) return null;
+
+  const CustomIcon = customIcons[mode]
+  if (CustomIcon) return React.cloneElement(CustomIcon, {...props})
+
+  return <BaseModeIcon mode={mode} {...props} />
+}
```

Comment:
I thought we were just going to get rid of customIcons in config.yml and use config.js? Is there any benefit to splitting up icon config between the two files?

Reply:
I saw this comment in `config.js` that made me back track (I have not tried to check it)
```
 * Caveats: This file cannot import the following:
 * - other files
 * - modules not installed in the node_modules of this project
```